miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/navigator/MiniAppExternalUrlLoader.kt

```diff
@@ -0,0 +1,48 @@
+package com.rakuten.tech.mobile.miniapp.navigator
+
+import android.app.Activity
+import android.content.Intent
+import com.rakuten.tech.mobile.miniapp.MiniAppScheme
+
+/**
+ * This support the scenario that external loader redirect to url which is only supported in mini app view,
+ * close the external loader and emit that url to mini app view by [ExternalResultHandler.emitResult].
+ * @param miniAppId The id of loading mini app.
+ * @param activity The Activity contains webview. Pass the activity if you want to auto finish
+ * the Activity with current external loading url as result data.
+ **/
+class MiniAppExternalUrlLoader(miniAppId: String, private val activity: Activity? = null) {
+
+    companion object {
+        const val returnUrlTag = "return_url_tag"
+    }
+
+    private val miniAppScheme = MiniAppScheme(miniAppId)
+
+    /**
+     * Determine to close the external loader.
+     * Use this in the return value of [WebViewClient.shouldOverrideUrlLoading(WebView, WebResourceRequest)].
+     **/
+    fun shouldOverrideUrlLoading(url: String): Boolean {
+        if (shouldClose(url)) {
+            closeExternalView(url)
+            return true
+        }
+
+        return false
+    }
+
+    /**
+     * In case you do not want to finish activity which contains webview automatically, use this to
+     * check should stop the external webview loader and send the current url to mini app view.
+     */
+    fun shouldClose(url: String): Boolean = miniAppScheme.isMiniAppUrl(url)
```

Comment:
We need this in case hostapps want to check the scenario themselves. Here is noted in [document.](https://github.com/rakutentech/android-miniapp/blob/4456720e8bfbf7ca125483448574a0c3addc8c8f/miniapp/USERGUIDE.md#manual-check-by-host-app).